velox/functions/prestosql/tests/StringFunctionsTest.cpp

```diff
@@ -2281,3 +2281,97 @@ TEST_F(StringFunctionsTest, concatInSwitchExpr) {
       {"This is a long sentence-zzz"_sv, "aaa-This is some other sentence"_sv});
   test::assertEqualVectors(expected, result);
 }
+
+TEST_F(StringFunctionsTest, fromBigEndian32) {
+  const auto fromBigEndian32 = [&](std::optional<std::string> arg) {
+    return evaluateOnce<int32_t, std::string>(
+        "from_big_endian_32(c0)", {arg}, {VARBINARY()});
+  };
+
+  EXPECT_EQ(std::nullopt, fromBigEndian32(std::nullopt));
+  EXPECT_THROW(fromBigEndian32(hexToDec("01")), VeloxUserError);
+  EXPECT_THROW(fromBigEndian32(hexToDec("0000000000000001")), VeloxUserError);
+
+  EXPECT_EQ(0, fromBigEndian32(hexToDec("00000000")));
+  EXPECT_EQ(1, fromBigEndian32(hexToDec("00000001")));
+  EXPECT_EQ(-1, fromBigEndian32(hexToDec("FFFFFFFF")));
+  EXPECT_EQ(12345678, fromBigEndian32(hexToDec("00BC614E")));
+  EXPECT_EQ(-12345678, fromBigEndian32(hexToDec("FF439EB2")));
+  EXPECT_EQ(2147483647, fromBigEndian32(hexToDec("7FFFFFFF")));
+  EXPECT_EQ(-2147483647, fromBigEndian32(hexToDec("80000001")));
+  EXPECT_EQ(-2147483648, fromBigEndian32(hexToDec("80000000")));
+}
+
+TEST_F(StringFunctionsTest, toBigEndian32) {
+  const auto toBigEndian32 = [&](std::optional<int32_t> arg) {
+    return evaluateOnce<std::string, int32_t>(
+        "to_big_endian_32(c0)", {arg}, {INTEGER()});
+  };
+
+  EXPECT_EQ(std::nullopt, toBigEndian32(std::nullopt));
+
+  EXPECT_EQ(hexToDec("00000000"), toBigEndian32(0));
+  EXPECT_EQ(hexToDec("00000001"), toBigEndian32(1));
+  EXPECT_EQ(hexToDec("FFFFFFFF"), toBigEndian32(-1));
+  EXPECT_EQ(hexToDec("00BC614E"), toBigEndian32(12345678));
```

Comment:
Please add more boundary values 2^32 and -(2^32).

Reply:
2^32 and -(2^32) are outside of int32_t range, this would just overflow into smaller numbers. I do test INT_MIN and INT_MAX in the tests.

Reply:
Yeah, they should overflow and it will be worthwhile to add a test to examine that behavior more closely anyways.

Reply:
Yes, I added overflow tests for the functions.